src/Kiota.Builder/Writers/Dart/CodeMethodWriter.cs

```diff
@@ -0,0 +1,878 @@
+            writer.WriteLine($"writer.{GetSerializationMethodName(otherProp.Type, method)}(null, {booleanValue}{otherProp.Name});");
+            writer.CloseBlock();
+            if (!includeElse)
+                includeElse = true;
+        }
+    }
+    private void WriteSerializerBodyForIntersectionModel(CodeMethod method, CodeClass parentClass, LanguageWriter writer)
+    {
+        var includeElse = false;
+        foreach (var otherProp in parentClass
+                                        .GetPropertiesOfKind(CodePropertyKind.Custom)
+                                        .Where(static x => !x.ExistsInBaseType)
+                                        .Where(static x => x.Type is not CodeType propertyType || propertyType.IsCollection || propertyType.TypeDefinition is not CodeClass)
+                                        .OrderBy(static x => x, CodePropertyTypeBackwardComparer)
+                                        .ThenBy(static x => x.Name))
+        {
+            var serializationMethodName = GetSerializationMethodName(otherProp.Type, method);
+            var booleanValue = serializationMethodName == "writeBoolValue" ? "value:" : "";
+            writer.StartBlock($"{(includeElse ? "else " : string.Empty)}if({otherProp.Name} != null) {{");
+            writer.WriteLine($"writer.{GetSerializationMethodName(otherProp.Type, method)}(null, {booleanValue}{otherProp.Name});");
+            writer.CloseBlock();
+            if (!includeElse)
+                includeElse = true;
+        }
+        var complexProperties = parentClass.GetPropertiesOfKind(CodePropertyKind.Custom)
+                                            .Where(static x => x.Type is CodeType propType && propType.TypeDefinition is CodeClass && !x.Type.IsCollection)
+                                            .ToArray();
+        if (complexProperties.Length != 0)
+        {
+            if (includeElse)
+            {
+                writer.StartBlock("else {");
+            }
+            var firstPropertyName = complexProperties.First().Name;
+            var propertiesNames = complexProperties.Skip(1).Any() ? complexProperties.Skip(1)
+                                .Select(static x => x.Name)
+                                .OrderBy(static x => x)
+                                .Aggregate(static (x, y) => $"{x}, {y}") : string.Empty;
+            var propertiesList = string.IsNullOrEmpty(propertiesNames) ? "" : $", [{propertiesNames}]";
+            writer.WriteLine($"writer.{GetSerializationMethodName(complexProperties.First().Type, method)}(null, {firstPropertyName}{propertiesList});");
+            if (includeElse)
+            {
+                writer.CloseBlock();
+            }
+        }
+    }
+
+    protected virtual void WriteCommandBuilderBody(CodeMethod codeElement, CodeClass parentClass, RequestParams requestParams, bool isVoid, string returnType, LanguageWriter writer)
```

Comment:
We should probably remove this function. Then throw exception in the caller directly in the switch case on line 123(and remove the irrelevant code there.)